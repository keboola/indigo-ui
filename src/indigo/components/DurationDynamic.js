import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { timeInWords, durationFrom } from './durationHelpers';

class DurationDynamic extends React.Component {
  constructor(props) {
    super(props);
    this.tick = this.tick.bind(this);
    this.state = {
      endTime: moment().format()
    };
  }

  componentDidMount() {
    this.interval = setInterval(this.tick, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  tick() {
    this.setState({
      endTime: moment().format()
    });
  }

  render() {
    return <span>{timeInWords(durationFrom(this.props.startTime, this.state.endTime))}</span>;
  }
}

DurationDynamic.propTypes = {
  startTime: PropTypes.string
};

export default DurationDynamic;
