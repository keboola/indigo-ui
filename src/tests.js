import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
