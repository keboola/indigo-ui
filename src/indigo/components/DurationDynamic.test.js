import React from 'react';
import moment from 'moment';
import { snapshot } from '../../tests';
import DurationDynamic from './DurationDynamic';

const fakeNow = '2018-11-26T15:11:05+0100';
const fakeNowMs = moment(fakeNow).valueOf();

global.Date.now = jest.fn(() => fakeNowMs);

const timePast = moment()
  .subtract(124, 'seconds')
  .format();

describe('<DurationDynamic />', () => {
  it('should render dynamic 2 min 4 sec', () => {
    snapshot(<DurationDynamic startTime={timePast} />);
  });
});
