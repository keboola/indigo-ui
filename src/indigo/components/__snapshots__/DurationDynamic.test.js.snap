// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DurationDynamic /> should render dynamic 2 min 4 sec 1`] = `
<span>
  2 min 4 sec
</span>
`;
